### DISCUSSION TITLE
How do I query a relationship attribute?

### DISCUSSION THREAD

**Discussion by afiqzu** [appwrite/appwrite#7369] [Q&A] [answered]
For example I have to collections 'Users' and 'Posts' with a two way many-to-many relationship of 'likes' and 'likedBy' respectively. How do I query the likes of a User? I have tried listDocuments with Query.select(['likes']) but I am getting error. Thanks in advance!

**Comment by stnguyen90** [comment, marked as answer]
Queries on relationships are not supported. That said, we do plan on adding support.

Source: https://appwrite.io/docs/products/databases/relationships#queries

**Reply by afiqzu** [reply]
Ok thanks!

**Reply by sencerburak** [reply]
Then what is even the point of having relations?

**Reply by hadnet** [reply]
More than a year! Jesus!

**Reply by sencerburak** [reply]
I long switched out of appwrite due to this. In my experience building my backend from scratch was easier than working around this  critical limitation

**Reply by yassir150** [reply]
finally the appwrite team add the relationship select query https://appwrite.io/docs/products/databases/queries#relationship-select

**Comment by m4jor101** [comment]
when is it going to be launched, this is such an essential query that should have been launched when relationships were introduced.

**Comment by soranoo** [comment]
Looking forward to its implementation! Will it be mentioned in the road map?

**Comment by mohaa007** [comment]
any updates?

**Comment by Harvindar994** [comment]
Is there any update

**Comment by stefanadelbert** [comment]
This is important for me too.

**Comment by hadnet** [comment]
WTF! This is ridiculous, guys!

**Comment by lukebrandonfarrell** [comment]
Is this still not supported?

We’re using Appwrite in a pretty complex application and decided to add search. It works when querying a one-to-many relationship, but we get an error when querying a many-to-many relationship. I see threads from over two years ago discussing this feature being added soon. If it still hasn’t landed, I’m seriously considering creating an abstract service layer and migrating all our applications from Appwrite to Supabase or Convex.

It’s a shame because I really liked Appwrite.

(cc @stnguyen90)

**Reply by lukebrandonfarrell** [reply]
Seriously we are holding out because we actually like you guys and the product, but this is really starting to become a blocker in all our projects, while it's a big effort to migrate it's something we are considering. Please give us a update on this at least to know if it's on the roadmap @stnguyen90 (cc @eldadfux). The silence on this is not helping.

**Reply by soranoo** [reply]
I moved to Convas for better type safely and granular query control.

**Reply by lukebrandonfarrell** [reply]
Oh yeah, I’m at my limit with this service now. We’re only still using it because we’ve invested a lot of time configuring it and don’t want to migrate our entire data layer at the moment. Seriously—after they launched Sites and a bunch of features no one asked for, while core functionality is still missing and a daily struggle. We’re already hitting limitations, and we’ve barely even launched the app.

**Comment by yyi0708** [comment]
This is important to us. Come on, maintainer

**Comment by Ninjonik** [comment]
This is a **critical missing feature**. Without proper relationship queries, the database is effectively unusable for any more complex application.

If Appwrite database was truly NoSQL rather than a NoSQL-style API on top of MariaDB, this limitation might be understandable. But given the fact that Appwrite uses MariaDB, then the lack of relationship queries is a serious bummer.  

Features like messaging, Appwrite sites, or updating counters are useful, but they are **not fundamental functionalities** in my opinion. Relationship queries are essential for any meaningful application, and notably, they *did* work (at least partially) in Appwrite version 1.5.5. For example, this query used to work:

```js
const res = await databases.listDocuments(database, "messages", [
      Query.equal("room", roomId),
      Query.orderDesc("$updatedAt"),
      Query.limit(50),
]);
```

I believe this feature should be prioritized over the others, and it seems I’m not the only one who thinks so.

**Comment by nox-plus** [comment]
In a world where even an enduser tool like Notion has proper databases and relations with queries, it's unacceptable for appwrite to still not support this. I'm seriously thinking of switching to directus.

**Comment by lukebrandonfarrell** [comment]
Plus, they’re pretty much ignoring this request. I’ve tried asking for a timeline in Discord, but I didn’t even get a response. We’re switching to Supabase now. Honestly, I was so excited to build with Appwrite when we first started our project, but a year later it’s been a complete shit show. The platform lacks data integrity, and I’m not sure how they decided to remove the Beta label when core functionality is still missing and there are so many critical issues. We’ve suffered multiple data losses through database restores and using the CLI.

**Comment by stefanadelbert** [comment]
I considered Appwrite, Directus and Supabase for a project about a year ago. I did POCs with all of them. Got quite far with Directus, but then rejected it based on their commercial model. Rejected Appwrite because of no support for Postgres and the problems with relationships. Settled on Supabase and I've been pretty happy.
I still prefer Firebase, but like that I can self-host Supabase. There are a few missing features (database migrations, etc.) in the community edition of Supabase, but it was still the right decision in my situation.

**Reply by lukebrandonfarrell** [reply]
Good to know you've have success with Superbase, that will be our next step to migrate.

**Comment by AbirSantra** [comment]
Any updates on this?

**Reply by Christophershivers** [reply]
It seems like they added querying for relationships 

https://appwrite.io/docs/products/databases/queries#relationship-select
